src/components/forms/ProfileCreationForm.js

```diff
@@ -39,11 +38,11 @@ function ProfileCreationForm(props) {
               fname, lname, birthdate, phonenum,
             } = values;
 
-            await initCreation(userId, fname, lname, birthdate, phonenum);
+            await updateProfile(userId, fname, lname, birthdate, phonenum);
```

Comment:
can you have asycn api call inside action, and don't manually set this, what if the api call failed, then you have stale data in ur redux store ?